compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -284,6 +284,22 @@ fn ios_lld_platform_version() -> String {
     format!("{major}.{minor}")
 }
 
+pub fn mac_catalyst_deployment_target() -> &'static str {
+    // Consider making this user-changeable in the future.
+    "14.0"
+}
+pub fn mac_catalyst_sdk_version() -> &'static str {
+    // Consider making this user-changeable in the future
+    "16.2"
+}
+pub fn mac_catalyst_llvm_target(arch: &str) -> &'static str {
+    // Catalyst requires LC_BUILD_VERSION information in object files.
+    // Seems that newer xcode refuses to compile for older than 14.0
+    // It returns a &'static str because TargetOptions::add_pre_link_args
+    // requires it.
+    if arch == "arm64" { "arm64-apple-ios14.0-macabi" } else { "x86_64-apple-ios14.0-macabi" }
```

Comment:
For llvm, I believe we can use this triple without specifying target os version number.

For example, `arm64-apple-ios14.0-macabi` can be `arm64-apple-ios-macabi`.

```shell
$ clang -target x86_64-apple-ios-macabi demo.c # this can work
```

Similar case: #106925